### PR TITLE
DOC: add note to docstring about `dtype` keyword when creating Series from abother Series

### LINKED ISSUE #59060: BUG: dtype keyword in Series constructor ignored for Series input in certain cases
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd

try:
    s = pd.Series([1, 2, 'x', 4, 5], dtype=int)
    print("1. No exception")
except Exception as ex:
    print("1. Exception thrown", ex)

s = pd.Series([1, 2, 'x', 4, 5])
try:
    s2 = pd.Series(s, dtype=int)
    print("2. No exception")
except Exception as ex:
    print("2. Exception thrown", ex)

print("s2 dtype", s2.dtype)
```

### Issue Description

The behavior of the above code doesn't seem right. 
If I try to create a Series with `dtype=int` using a list that contains a string, it throws a ValueError, as expected.

But if the input to the Series constructor is another Series that contains a string, there is no exception and it quietly returns a new Series with `dtype=object`. 

Is this expected? If so it's very counterintuitive. Perhaps the recommended way is 
`s.astype(int)`
but it seems like this should produce the same behavior

### Expected Behavior

In both of the cases above, it should throw a ValueError. Instead, only the first case does:

```
1. Exception thrown invalid literal for int() with base 10: 'x'
1. No exception
s2 dtype object
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.11.6.final.0
python-bits           : 64
OS                    : Darwin
OS-release            : 23.1.0
Version               : Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : None
LANG                  : en_US.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 2.2.2
numpy                 : 1.23.4
pytz                  : 2022.5
dateutil              : 2.8.2
setuptools            : 65.5.0
pip                   : 24.0
Cython                : None
pytest                : None
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : None
html5lib              : None
pymysql               : None
psycopg2              : None
jinja2                : 3.1.2
IPython               : 8.12.0
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.2
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : None
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : 3.1.2
pandas_gbq            : None
pyarrow               : None
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.13.1
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2023.3
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by taranarmo**
When creating Series from another Series `dtype` keyword is not not used. Add note on `dtype` keyword to Series and DataFrame constructor. See GH #59060.

- [x] closes #59060 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by mroeschke — CHANGES_REQUESTED** [review]
A warning would be too noisy. A note in the docs would be sufficient.

**Comment by taranarmo** [comment]
Undo the previous changes and add a note to the `dtype` keyword on `pd.Series` and `pd.DataFrame` that says that the keyword is ignored if `data` is Series and DataFrame correspondingly.

File: pandas/core/frame.py [hunk 1]
```diff
@@ -531,6 +531,7 @@ class DataFrame(NDFrame, OpsMixin):
         will perform column selection instead.
     dtype : dtype, default None
         Data type to force. Only a single dtype is allowed. If None, infer.
+        If `data` is DataFrame then is ignored.
```
Comment (mroeschke) [inline pandas/core/frame.py:534]:
```suggestion
        If ``data`` is DataFrame then is ignored.
```

File: pandas/core/series.py [hunk 2]
```diff
@@ -256,6 +256,7 @@ class Series(base.IndexOpsMixin, NDFrame):  # type: ignore[misc]
         Data type for the output Series. If not specified, this will be
         inferred from `data`.
         See the :ref:`user guide <basics.dtypes>` for more usages.
+        If `data` is Series then is ignored.
```
Comment (mroeschke) [inline pandas/core/series.py:259]:
```suggestion
        If ``data`` is Series then is ignored.
```

File: pandas/core/frame.py (hunk 1 above)
Comment (taranarmo) [reply pandas/core/frame.py:534]: Why do we need double backticks if elsewhere they are single?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @taranarmo

**Comment by jorisvandenbossche** [comment]
I was going to comment that I find this a bit surprising behaviour, and so whether we shouldn't rather consider this as a bug instead of documenting it. 

But from a quick check, it seems that the `dtype` argument in fact is _not_ ignored, at least not in all cases (https://github.com/pandas-dev/pandas/issues/59060 clearly shows there are also cases where it is indeed ignored):

```
In [18]: pd.Series(pd.Series([1.0], dtype=float), dtype=int)
Out[18]: 
0    1
dtype: int64
```

Here it does cast the float series to an int64 dtype, so the `dtype` keyword was not ignored.

Will also comment on the issue, but so it seems the simple "If ``data`` is Series then is ignored." message added in this PR is not fully correct.

**Comment by jbrockmendel** [comment]
IIUC I agree with Joris. The solution is not to change the doc, but to make the behavior consistent. The dtype kwarg should never be silently ignored.

**Pull request merged**